bootloader/utils/dirlist.c

```diff
@@ -40,7 +40,7 @@ char *dirlist(const char *directory, const char *pattern)
 			res = f_readdir(&dir, &fno);
 			if (res || !fno.fname[0])
 				break;
-			if (!(fno.fattrib & AM_DIR))
+			if (!(fno.fattrib & AM_DIR) && (includeHiddenFiles || !(fno.fattrib & AM_HID && fno.fname[0] != '.')))
```

Comment:
Make `fno.fname[0] != '.'`  always required.
There's no reason to even parse these imo.

```suggestion
			if (!(fno.fattrib & AM_DIR) && (fno.fname[0] != '.') && (includeHiddenFiles || !(fno.fattrib & AM_HID)))
```